File: AndBible/src/net/bible/service/format/osistohtml/TransChangeHandler.java
package net.bible.service.format.osistohtml;

import net.bible.service.common.Logger;

import org.xml.sax.Attributes;

/** Paragraph <p>...</p>
 * 
 * @author Martin Denham [mjdenham at gmail dot com]
 * @see gnu.lgpl.License for license details.<br>
 *      The copyright to this program is held by it's author. 
 */
public class TransChangeHandler {

	private HtmlTextWriter writer;
	
	@SuppressWarnings("unused")
	private OsisToHtmlParameters parameters;
	
	@SuppressWarnings("unused")
	private static final Logger log = new Logger("LHandler");

	public TransChangeHandler(OsisToHtmlParameters parameters, HtmlTextWriter writer) {
		this.parameters = parameters;
		this.writer = writer;
	}
	
	public String getTagName() {
        return "transChange";
    }

	public void start(Attributes attrs) {
		writer.write("<span class='transChange'>");
	}

	public void end() {
		writer.write("</span>");
	}
}

File: app/src/main/java/net/bible/service/format/osistohtml/taghandler/TransChangeHandler.kt
/*
 * Copyright (c) 2020 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
 *
 * This file is part of And Bible (http://github.com/AndBible/and-bible).
 *
 * And Bible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with And Bible.
 * If not, see http://www.gnu.org/licenses/.
 *
 */
package net.bible.service.format.osistohtml.taghandler

import net.bible.service.common.Logger
import net.bible.service.format.osistohtml.HtmlTextWriter
import net.bible.service.format.osistohtml.OSISUtil2
import net.bible.service.format.osistohtml.OsisToHtmlParameters
import org.xml.sax.Attributes

/** Paragraph
 *
 *...
 *
 * @author Martin Denham [mjdenham at gmail dot com]
 */
class TransChangeHandler(private val parameters: OsisToHtmlParameters, private val writer: HtmlTextWriter) : OsisTagHandler {
    override val tagName: String
        get() = OSISUtil2.OSIS_ELEMENT_TRANSCHANGE

    override fun start(attrs: Attributes) {
        writer.write("<span class='transChange'>")
    }

    override fun end() {
        writer.write("</span>")
    }

    companion object {
        private val log = Logger("LHandler")
    }

}
